app/v1/controllers/actions/AuthAction.scala

```diff
@@ -27,13 +26,25 @@ import v1.models.requests.IdentifierRequest
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class AuthAction @Inject()(override val authConnector: AuthConnector,
-                           val parser: BodyParsers.Default
-                          )(implicit val executionContext: ExecutionContext)
-  extends ActionBuilder[IdentifierRequest, AnyContent] with ActionFunction[Request, IdentifierRequest] with AuthorisedFunctions {
+trait AuthActionBase extends ActionBuilder[IdentifierRequest, AnyContent] with ActionFunction[Request, IdentifierRequest] with AuthorisedFunctions
```

Comment:
Whats the benefit to moving the logic that was in the extends out to a trait?

Reply:
@AdamCoops By creating a trait that extends all of those other traits instead of the class doing it means that you can use it as a super type for two classes that need to share all of the same extended traits.

Where previously we needed only a single AuthAction that performed exactly like an Action[Request, IdentifierRequest] with AuthorisedFunctions, we now need two. So by moving all of that shared functionality to a separate trait you remove duplication by simply extending both of the new classes by your new trait.

But even more interesting for us is that you can now use this trait as a supertype in variable/parameter types. So we can say that a method needs an "AuthActionBase" or that a method will return an "AuthActionBase" but it'll actually allow us to return either of our classes which is exactly what Juan's new AuthActionProvider is doing.

It also means that when our provider returns either of those actions, everything knows that no matter which class is returned it has X amount of functionality which is guaranteed to be shared by the two.